bindings/java/src/main/java/org/openucx/jucx/ucp/UcpRequest.java

```diff
@@ -49,7 +58,18 @@ public long getSenderTag() {
      * @return whether this request is completed.
      */
     public boolean isCompleted() {
-        return (getNativeId() == null) || isCompletedNative(getNativeId());
+        if ((getNativeId() != null) && isCloseRequest) {
+            updateRequestStatus(getNativeId());
+        }
+
+        return (status != UcsConstants.STATUS.UCS_INPROGRESS);
```

Comment:
getStatus

Reply:
It's safe to use the status field, since it's class private. `getStatus` will be inlined by JIT compiler at some point, but since this method is usually used in request progress loop, better to not rely on jit, IMO.

Reply:
ok